lib/IRGen/GenType.cpp

```diff
@@ -1114,6 +1116,30 @@ TypeConverter::getLegacyTypeInfo(NominalTypeDecl *decl) const {
   return found->second;
 }
 
+static const char *platformsWithLegacyLayouts[][2] = {
```

Comment:
Nitpick: this can use `llvm::StringLiteral`, and then it'll be faster to compare.

Reply:
It's only compared once per IRGenModule creation so I don't think it's performance-sensitive, but I'll see if it cleans it up anyway.